ios/Fabric/RNCSafeAreaProviderComponentView.mm

```diff
@@ -123,6 +124,7 @@ - (void)prepareForRecycle
   _currentSafeAreaInsets = UIEdgeInsetsZero;
   _currentFrame = CGRectZero;
   _initialInsetsSent = NO;
+  [NSNotificationCenter.defaultCenter removeObserver:self];
```

Comment:
Do we not need to re-add our listeners at some point?

Reply:
@jacobp100 Oh right. I missing this part. I am adding this now
Thanks!

Reply:
@jacobp100 I updated the code. Please take a look.